### ISSUE TITLE
Mismatch between number of axes in coordinate frames and transforms for NIRCAM spectral WCS. V2V3 units

### ISSUE THREAD

**Issue by mcara** [spacetelescope/jwst#5746]
1. In `jwst.assign_wcs.nircam.tsgrism` new axes are being added  to the transformations from the imaging mode (see `& Identity(2)`) but the number of axes in the coordinate frames are left unchanged (imaging frames are used without change). Unit tests for this mode should test whether the WCS' forward transformation works (i.e., `pixel_to_world()` which would use frames). Currently it is checking only that required frame names are present in WCS pipelines (I think).
https://github.com/spacetelescope/jwst/blob/8286294e82ec9be6c6979b649885f90bdf42f1be/jwst/assign_wcs/nircam.py#L254-L269

2. In the same function, units for the `v2v3` coordinate frame are `u.deg` while for all other instruments it seems to be set to `u.arcsec`:
https://github.com/spacetelescope/jwst/blob/8286294e82ec9be6c6979b649885f90bdf42f1be/jwst/assign_wcs/nircam.py#L178

3. A similar issue to `#1` (transforms with more coordinates than axes in frames), is present in `jwst.assign_wcs.nircam.wfss`:
https://github.com/spacetelescope/jwst/blob/8286294e82ec9be6c6979b649885f90bdf42f1be/jwst/assign_wcs/nircam.py#L393-L395

4. Also I have noticed that sometimes axes for the `v2v3` frame are called (labeled) `['V2', 'V3']` and sometimes `['v2', 'v3']`. This is probably not important now but it will make life more difficult to the users of these WCSes if axes names are not consistent.

**Comment by sosey** [comment]
I'm not sure I see the issue yet @mcara. This is due to the needs of the slitless spec modes.

TSO is extra special in that crpix1, crpix2 are always the location of the object being observed, and those match the ra,dec, so those items can be pinned in the transform chain.

However, the slitless transforms in general require more information than axes in the image, from end to either end:
`(xdetector, ydetector, xsource, ysource, order) <-> (ra, dec, wavelength, order)`

The same is true for WFSS mode.

**Comment by mcara** [comment]
The issue is when you try to run forward transformation with units like this:
```python
wcsobj(xd, yd, xs, ys, order, with_units=True)
```
or using APE 14 functions:
```python
 wcsobj.pixel_to_world(xd, yd, xs, ys, order, with_units=True)
```
I do not think this works. Without units only forward model is used. With units, info from frames is used to construct `SkyCoord` objects.

**Comment by sosey** [comment]
ah, the units was added after I last worked on this, if @nden doesn't have time in the next couple weeks I can come back and look deeper

**Comment by hbushouse** [comment]
> TSO is extra special in that crpix1, crpix2 are always the location of the object being observed, and those match the ra,dec, so those items can be pinned in the transform chain.

crpix1, crpix2 are so .... 20th century. JWST spectral data products no longer include any FITS WCS keywords. The keywords you're looking for are XREF_SCI, YREF_SCI.

**Comment by nden** [comment]
@stscijgbot-jp

**Comment by stscijgbot-jp** [comment]
_This issue is tracked on JIRA as [JP-1923](https://jira.stsci.edu/browse/JP-1923)._

**Comment by sosey** [comment]
@hbushouse yes, that's right...I was trying to point him towards the fact that aperture point reference for TSO is always used as the source object location :)

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-1923?focusedCommentId=510872#comment-510872):_

Fixed units (deg->arcsec) and 'v2', 'v3' axes names in #5765 (takes care of #&#x2060;2, #&#x2060;4).

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1923?focusedCommentId=511568#comment-511568):_

Fixed by #5771
